### PR TITLE
New 16.1-17.0 sql patch

### PR DESCRIPTION
**Pull request by kongtiaowang**
Release patch for upgrading from the most recent minor release (16.1) to Loris 17.0.

Note that projects upgrading from 16.0 should first apply the 16.0 to 16.1 upgrade patch.
[CR]

### CODE REVIEW

File: SQL/Release_patches/16.1_To_17.0_upgrade.sql [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+            ('xml'),
+            ('txt'),
+            ('nii'),
+            ('nii.gz'),
+            ('nrrd');
+
+ALTER TABLE `mri_processing_protocol` MODIFY `FileType` VARCHAR(255) DEFAULT NULL;
+ALTER TABLE `mri_processing_protocol` ADD FOREIGN KEY (`FileType`) REFERENCES `ImagingFileTypes`(`type`);
+ALTER TABLE `files` MODIFY `FileType` VARCHAR(255) DEFAULT NULL;
+ALTER TABLE `files` ADD FOREIGN KEY (`FileType`) REFERENCES `ImagingFileTypes`(`type`);
```
Comment (gluneau) [inline SQL/Release_patches/16.1_To_17.0_upgrade.sql]: Missing update statement for ongoing studies.  IBIS will be using this:

`UPDATE files SET FileType = SUBSTRING_INDEX(File,'.',-1) WHERE FileType = ''; `

File: SQL/Release_patches/16.1_To_17.0_upgrade.sql [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  `issueID` int(11) unsigned NOT NULL,
+  PRIMARY KEY (`userID`,`issueID`),
+  KEY `fk_issues_watching_2` (`issueID`),
+  CONSTRAINT `fk_issues_watching_1` FOREIGN KEY (`userID`) REFERENCES `users` (`UserID`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+UPDATE files_qcstatus as fqc SET fqc.Selected='true' WHERE fqc.Selected <> '';
+ALTER TABLE files_qcstatus CHANGE `Selected` `Selected` enum('true','false') DEFAULT NULL; 
+-- Add ScannerCandID column to files, back-populate the newly added column from parameter_file table, then add foreign key constraints 
+ALTER TABLE files ADD `ScannerID` int(10) unsigned default NULL;
+CREATE TEMPORARY TABLE ScannerIDs AS SELECT pf.FileID, pf.Value AS ScannerID FROM parameter_file AS pf LEFT JOIN parameter_type AS pt ON pf.ParameterTypeID=pt.ParameterTypeID WHERE pt.Name='ScannerID';
```
Comment (gluneau) [inline SQL/Release_patches/16.1_To_17.0_upgrade.sql]: Existing study will need to check if they have the permission to create temporary tables for the user id they will import this patch with.

Comment (christinerogers) [inline SQL/Release_patches/16.1_To_17.0_upgrade.sql]: Great point Greg.  let's add this note in the Release notes for Existing Projects.

File: SQL/Release_patches/16.1_To_17.0_upgrade.sql (hunk 1 above)
Comment (christinerogers) [inline SQL/Release_patches/16.1_To_17.0_upgrade.sql]: @gluneau  - which patch was this in? Or should we commit a further patch?

**Pull request merged**